### PR TITLE
Improve tests for SDK-based util functions, add tests to the new plugin framework equivalents

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR:

- Adds unit tests for the `getProject` function that depends on the provider SDK
- Adds unit tests for the `getProjectFramework` function, a version that that depends on the plugin framework instead of the SDK
- Refactors existing unit tests for util functions:
    - `TestGetRegion`
    - `TestGetZone`
    - Made one pre-existing test use sub-tests
    - Moved the location and name of another pre-existing test

---

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/utils/provider_test.go.erb
Comment (SarahFrench) [inline mmv1/third_party/terraform/utils/provider_test.go.erb:1]: I moved this test to a different file as it's not really a 'provider' test, though that may have made sense at the time this code was written!

**Comment by SarahFrench** [comment]
👆 That's the last little tweak from me, I learned a better way to omit a value in a framework test

**Review by zli82016 — APPROVED** [review]
LGTM. Please rebase the main branch because of the merged PR https://github.com/GoogleCloudPlatform/magic-modules/pull/7446. Thanks.

**Comment by SarahFrench** [comment]
I'll look into the failure later today

**Comment by SarahFrench** [comment]
Seems like the issue is that my test was using something that has since been removed from TPGB, very confusing

**Review by zli82016 — APPROVED** [review]

**Pull request merged**